### PR TITLE
Updates Kotlin and kotlinx.serialization dependencies to 1.4.0-rc

### LINKED ISSUE #25: Kotlin Serialization 1.0-M1 compatibility
I'm not sure if it's worth making an issue but here we go:  
With the new release of Kotlin 1.4.0-rc, Kotlin Serialization 1.0-M1-1.4.0-rc must be used, which bring some error when using  retrofit2-kotlinx-serialization-converter.

Those errors should be easily displayed once the project is updated but one example : 

```
    java.lang.NoClassDefFoundError: Failed resolution of: Lkotlinx/serialization/JvmResolvingKt;
        at com.jakewharton.retrofit2.converter.kotlinx.serialization.Factory.requestBodyConverter(Factory.kt:29)
```

Which is caused by the use of `serializerByTypeToken` which should be replaced as the `@Deprecated` say by `serializer(type)`.
There might be more.

I'm not sure if you'd rather wait for a serialization 1.0 release or make an intermediate version for the 1.0-M1? (as I don't know how much time there will be between 1.0-M1 and 1.0)

### PR DESCRIPTION
**Pull request by NitroG42**
Fixes #25. Closes #28.

### CODE REVIEW

**Comment by NitroG42** [comment]
I forgot to turn my formatter off, I'll do that and resubmit

**Comment by NitroG42** [comment]
2 spaces indentation = two force-push.
Should be good now!

File: build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buildscript {
   ext.versions = [
-    'kotlin': '1.3.70',
-    'serializationRuntime': '0.20.0',
+    'kotlin': '1.4.0-rc',
+    'serializationRuntime': '1.0-M1-1.4.0-rc',
```
Comment (wbervoets) [inline build.gradle:4]: @NitroG42 When using Kotlin 1.4.0 I have used org.jetbrains.kotlinx:kotlinx-serialization-core:1.0.0-RC as a dependency. (runtime jar was renamed to core).
When using your fork, I also had to exclude org.jetbrains.kotlinx:kotlinx-serialization-runtime and org.jetbrains.kotlinx:kotlinx-serialization-runtime-jvm from being included to fix duplicate class definitions.

Comment (NitroG42) [reply build.gradle:4]: I'm probably going to rewrite the pull requests for the latest change, and I also saw that 1.4.0, while out, still don't have a blog post announcement (which probably means that it's not officialy out). I wonder if they are waiting for serialization (and maybe others kotlinx library) to be released in 1.0.

Comment (abyrnes) [reply build.gradle:4]: Nope! Looks like you just had to wait a few hours:
https://blog.jetbrains.com/kotlin/2020/08/kotlin-1-4-released-with-a-focus-on-quality-and-performance/

**Comment by NitroG42** [comment]
I updated it to kotlin 1.4.0 + kotlin serialization 1.4.0-rc

**Comment by b95505017** [comment]
@JakeWharton Any possible to merge this?

**Comment by JakeWharton** [comment]
I will merge whenever I have time to adequately review. Since no one is paying me, you unfortunately have little control over when that is!

**Comment by JakeWharton** [comment]
Thankfully for you, today was a day of meetings and I had already spent the requisite time and was literally _minutes_ from pushing up my changes.

**Pull request merged**